src/pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -522,12 +524,23 @@ def _get_cache(result_metric: _ResultMetric, on_step: bool) -> Optional[Tensor]:
             cache = result_metric._forward_cache
         elif not on_step and result_metric.meta.on_epoch:
             if result_metric._computed is None:
-                # always reduce on epoch end
                 should = result_metric.meta.sync.should
-                result_metric.meta.sync.should = True
+                if not result_metric.meta.sync.should and distributed_available():
+                    if _fault_tolerant_training():
+                        # make sure to always sync across devices when fault-tolerant is used
+                        result_metric.meta.sync.should = True
+                    else:
```

Comment:
I think this should be
```suggestion
                    elif distributed_available():
```

Reply:
distributed check is required for fault tolerance too.